jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -337,7 +337,7 @@ def check_eq(a, b, c, msg):
         f"Bias must have same seq length as QKV, got {bT} and {bS}")
 
   # check q_seqlen/kv_seqlen/q_offsets/kv_offsets
-  expected_rank = 2 if q_offsets is not None else 1
+  expected_rank = 1
```

Comment:
@Cjkkkk Can you comment on this change? I feel like I am wrong here, but I needed to make this change to pass the e2e test, and I am confused about the original intention.